### PR TITLE
docs: add post about learning quickly

### PR DESCRIPTION
**Pull request by jsjoeio**
# New Article: "How to Learn Quickly"

_Context: I have a book idea around learning things quickly. I tested it first with a [tweet](https://twitter.com/jsjoeio/status/1265026417282043905?s=20). That received positive feedback so now I've taken that and applied to it to something (learning TypeScript) which is this article._ 

For this article and validating my idea, I will measure:
- article views
- email sign ups 

Remaining todos:
- [x] get feedback
- [x] fix excalidraw errors
- [x] address Sean's feedback
- [x] address swyx's feedback
- [x] address Prince's feedback
- [x] add thank you section
- [ ] check site preview
- [ ] add reviewers to contributors table
- [ ] add ConvertKit form for people to sign up to get PDF

### 💻 [Site Preview](https://deploy-preview-165--keen-roentgen-cafcb7.netlify.app/learn-quickly)

Date to publish: **Friday, July 3rd** 🤞🏼 (if all goes well)

### CODE REVIEW

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
+---
+slug: 'learn-typescript-quickly'
+date: '2020-06-19'
+title: 'How to Learn Typescript Quickly'
+description: 'In the post, I show you everything you need to do to learn TypeScript quickly.'
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
description: 'In the post, I show you everything you need to do to learn TypeScript quickly using the Fast Framework.'
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
+---
+slug: 'learn-typescript-quickly'
+date: '2020-06-19'
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
date: '2020-07-03'
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+title: 'How to Learn Typescript Quickly'
+description: 'In the post, I show you everything you need to do to learn TypeScript quickly.'
+tagline: 'The only guide you need'
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
tagline: 'Using the Fast Framework'
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+published: true
+---
+
+A few months back, I set out a goal to learn TypeScript deeply in 3 months. And [I did it](https://twitter.com/jsjoeio/status/1250490760362774529)! Since then, others have asked how they can do the same. This guide will provide you with a framework. It's called the **Fast Framework**.
+
+Imagine you're in Mario Kart and racing on a new map. You will only have achieved your goal if you cross the finish line. Taking this analogy, the framework can be broken into three phases:
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
Imagine you're in Mario Kart and racing on a new map. Your goal is to cross the finish line. Taking this analogy, we break the framework into three phases:
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+- Make your Map - Vision / Timeframe / Timeline / Organization
+- Stay on Track -  Accountability / Momentum / Sharing / Focus
+- Find the Flag - Practice / Confidence / Vocabulary / Depth
+
+In each phase, I'll cover different aspects that will help you learn fast and reach your goal. The first phase will cover the foundation for your learning project. Think of this as vision, timeframe, timeline, and organization. In the phase after, the focus on staying on track and I'll discuss aspects related to accountability, momentum, sharing, and focus. Lastly, I'll help you find the flag and talk about practice, confidence, vocabulary, and depth in the final phase of the journey.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
In each phase, I'll cover different aspects that will help you learn fast and reach your goal. I'll use learning TypeScript as the example. The first phase will cover the foundation for your learning project. Think of this as vision, timeframe, timeline, and organization. In the phase after, the focus  will be staying on track and I'll discuss aspects related to accountability, momentum, sharing, and focus. In the final phase of the journey, I'll help you find the flag. There, I'll talk about practice, confidence, vocabulary, and depth.
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+- Read the [Official TypeScript Handbook](https://www.typescriptlang.org/docs/handbook/basic-types.html)
+
+- Produce small code examples for all concepts
+
+- Contribute to TypeScript, [the language](https://github.com/microsoft/TypeScript/issues?q=is%3Aopen+is%3Aissue+label%3A"good+first+issue")
+- Contribute to [typescript-cheatsheets/react-typescript-cheatsheet](https://github.com/typescript-cheatsheets/react-typescript-cheatsheet)
+
+**Don't rush through this part**. Spend some time thinking about your objective and what you need to achieve to get there. Completing your key results should allow you to confidently say you've completed your objective.
+
+With this vision in mind, you now have clarity on it means when you say "learn TypeScript." and you know what outcomes you expect to achieve.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
With this vision in mind, you now have clarity on it means when you say "learn TypeScript" and you know what you expect to achieve.
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+
+### Establish a Timeframe
+
+You don't want to be racing on this map forever. You need to be realistic and ask yourself how much time we have for this. Estimating time to complete tasks in software is hard. We'll try our best by using something I like to call the **Timeframe Formula**:
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
You don't want to be racing on this map forever. You need to be realistic and ask yourself how much time you have for this. Estimating time to complete tasks is hard. You can use what I like to call the **Timeframe Formula**:
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+
+https://excalidraw.com/#json=4877101601652736,SeRw5x2aEzwtztaHHk0xUQ
+
+For example, when I did a similar project, I gave myself 3 months with 8 hours per week. Using the Timeframe Formula, you get:
+
+(3 months * 4 weeks/month) * 8 hours/week = 96 hours total.
+
+https://excalidraw.com/#json=5134809571524608,ASGbR83wxxoafOuInmZ5Ew
+
+You now have your timeframe and expectation of how much you plan to dedicate each week.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
You now have your timeframe. Use this to set expectations on how much time you plan to dedicate each week.
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+
+### Build a Timeline
+
+The next thing we need to add to our map is a timeline. Think of this as a week by week calendar. It helps you think less because you have already decided what to do for that week. It won't be perfect. You may need to readjust, but at least you will have established the framing.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
The next thing you need to add to you map is a timeline. Think of this as a week by week calendar. It helps you think less because you will already know what to do for that week. It won't be perfect. You may need to readjust, but you will have removed friction.
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+    - [ ] Chapter 1 - Introduction
+    - [ ] Chapter 2 - TypeScript: a 10_000 foot view
+      - [ ] Book exercises
+    - [ ] Chapter 3 - All About Types
+      - [ ] Book exercises
+  - [ ] Official TypeScript Handbook (OTH)
+    - [ ] Basic Types
+    - [ ] Variable Declarations
+
+You know yourself best. Think back to the Timeframe Formula. The example estimated 8hrs/week and 12 weeks in total. Knowing the speed that you read it, you would adjust accordingly. A complete timeline would have all the weeks (i.e. 12 in this case) filled out.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
You know yourself best. Think back to the Timeframe Formula. The example estimated 8hrs/week and 12 weeks in total. Knowing how fast you work (i.e. read, code, etc.), you would adjust accordingly. A complete timeline would have all the weeks (i.e. 12 in this case) filled out.
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+
+Two important tips:
+
+1. Leave 1-2 weeks buffer towards the end
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]: 12
```suggestion
1. Leave ~10%(1-2 weeks) buffer towards the end
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+2. Give yourself some breathing room
+
+Life comes up and you may fall behind. That's where the buffer comes in! Some weeks, you may find that you complete everything in 6 hours. No need to fill those two other hours if you don't want to. Reward yourself with rest.
+
+With your timeline, you'll be good to go!
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
With a timeline, you are be good to go!
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 13]
```diff
@@ -0,0 +1,259 @@
...
+
+### Keep it Organized
+
+You want to stay organized on this journey. That means a place to keep track of your timeline, log any notes or things you learn, and have visual indicators for the progress you're making.
+
+I use Notion for this because it supports this type of project management well. I make a place for the project. I add the timeline and use checkboxes. Then, I check them off as I go and take notes as I learn new things.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
I use Notion for this because it supports project management well. I make a place for the project. I add the timeline and use checkboxes. Then, I check them off as I go and take notes as I'm learning.
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 14]
```diff
@@ -0,0 +1,259 @@
...
+
+Here is a list of tools you might want to check out:
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
Here is a non-exhaustive list of tools you might want to check out:
```
```suggestion
Here is a list of tools you might want to check out:
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 15]
```diff
@@ -0,0 +1,259 @@
...
+
+## Stay on Track
+
+Now that we have our map, we are ready for the next phase: stay on track.
+
+In this phase, the goal is to focus on developing the systems that will help you stay accountable, build momentum, share with others and, stay focused.
+
+### Keep Yourself Accountable
+
+You always want someway to stay accountable for your project. Think of this as your race crew. They make sure you're fueled, your tires are rotated, you're hydrated, and you're ready to finish the race. We want a similar type of support and accountability.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
You always want someway to stay accountable for your project. Think of this as your race crew. They make sure you're fueled, your tires are rotated, you're hydrated, and you're ready to finish the race. We want a similar type of support and accountability.
```
```suggestion
You always want someway to stay accountable for your project. Think of this as your race crew. They make sure you're fueled, your tires are rotated, you're hydrated, and you're ready to finish the race. We want a similar type of support and accountability.
```
```suggestion
You always want to stay accountable for your project. Think of this as your race crew. They make sure you're fueled, your tires are rotated, you're hydrated, and you're ready to finish the race. We want a similar type of support and accountability.
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 16]
```diff
@@ -0,0 +1,259 @@
...
+
+The other thing you might want to try is [**deep work**](https://www.calnewport.com/books/deep-work/). An idea by Cal Newport, it's where you create an environment that lets you focus on that project for a certain amount of time (such as 1 hour or more) without distractions. Figure out when you're most productive. For me, it's in the morning before work. I wake up early, go through my routine, and then spend some time learning.
+
+Lastly, find tools to help you stay focused. I am a big fan of the **Pomodoro Technique**. I have an app called [Stretchly](https://hovancik.net/stretchly/) that reminds me to take a 20-second break every 20 minutes and a 5-minute break every 40 minutes. This allows me to focus for that period and then rest. It then repeats itself like a cycle.
+
+## Find the Flag
+
+The last and final leg of this learning journey is finding the flag. You want to make it to the finish line and feel accomplished when you get there. Here, you want to use systems that will help you practice, build confidence, remember vocabulary, and go for depth so that you find the checkered flag.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
The last and final leg of this learning journey is finding the flag. You want to make it to the finish line and feel accomplished when you get there. Here, you want to use systems that will help you practice, build confidence, remember vocabulary, and go for depth so that you find the checkered flag.
```
```suggestion
The last and final leg of this learning journey is finding the flag. You should make it to the finish line and feel accomplished when you get there. Here, you want to use systems that will help you practice, build confidence, remember vocabulary, and go for depth. This will lead to the beloved checkered flag.
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 17]
```diff
@@ -0,0 +1,259 @@
...
+
+### Practice
+
+Learning means nothing without practice. With your new TypeScript knowledge, you can build projects or contribute to open source. This is where the application phase of the learning cycle appears. It doesn't matter what you build, but that the act of building happens.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]: Maybe add a diagram of the learning cycle?

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 18]
```diff
@@ -0,0 +1,259 @@
...
+
+When I was learning, I followed a tutorial to [build a tiny compiler](https://github.com/jamiebuilds/the-super-tiny-compiler). It was written in JavaScript,  but I did it in TypeScript. This forced me to figure out things on my own.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
When I was learning, I followed a tutorial to [build a tiny compiler](https://github.com/jamiebuilds/the-super-tiny-compiler). It was written in JavaScript, but I did it in TypeScript. This forced me to figure out things on my own.
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 19]
```diff
@@ -0,0 +1,259 @@
...
+
+Other things I did include contributing to open source. Remember our key results? Here's two of them that apply here:
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
Another idea is contributing to open source. Remember our key results? Here's two of them that apply here:
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 20]
```diff
@@ -0,0 +1,259 @@
...
+
+- Contribute to TypeScript, [the language](https://github.com/microsoft/TypeScript/issues?q=is%3Aopen+is%3Aissue+label%3A"good+first+issue")
+- Contribute to [typescript-cheatsheets/react-typescript-cheatsheet](https://github.com/typescript-cheatsheets/react-typescript-cheatsheet)
+
+This is where that happens if it hasn't already! The issues that you help with may be minor, but that's not the point.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
This is where that happens if it hasn't already. The issues that you help with may be minor, but that's not the point.
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 21]
```diff
@@ -0,0 +1,259 @@
...
+
+**Practice is where you learn the most**.
+
+### Build Confidence
+
+You want to build confidence in whatever you learning. This will make you feel comfortable using your new knowledge.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
You want to build confidence in whatever you're learning. This will make you feel comfortable using your new knowledge.
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 22]
```diff
@@ -0,0 +1,259 @@
...
+- teaching others
+- sharing (as we said before)
+- reviewing what you've learned
+- helping others
+
+### Don't Skip Vocab
+
+Vocabulary is often overlooked. Don't skip out on it! It helps build confidence and it will solidify what you know. It's also a great tool to fight imposter syndrome. Knowing the vocabulary will allow you to build **expertise**.
+
+To learn and remember vocabulary, use a *spaced-repetition system* such as [Anki](https://apps.ankiweb.net/). I used it when I was studying TypeScript. You can use it to remember things such as:
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
To learn and remember vocabulary, use a *spaced-repetition system* such as [Anki](https://apps.ankiweb.net/). I used it with my TypeScript learning project. You can use it to remember things such as:
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 23]
```diff
@@ -0,0 +1,259 @@
...
+
+- terminology
+- concepts
+- patterns
+- syntax
+
+### Go for Depth
+
+The last thing in this phase is going for depth. Deep learning. Deep understanding.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
The last thing in this phase is going for depth. **Deep learning leads to deep understanding**.
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 24]
```diff
@@ -0,0 +1,259 @@
...
+
+There are two techniques you can use to deepen what you learn. The first is called the [Feynman Technique](https://fs.blog/2012/04/feynman-technique/). Take a concept you are learning, such as [type inference in TypeScript](https://www.typescriptlang.org/docs/handbook/type-inference.html) and explain it as you would to a seven-year-old.
+
+The idea is that if you can't do that, you don't know the concept well enough.
+
+Note: the age of the person towhom you're explaining your learning concept can vary. The goal is for you to be able to explain something you understand to someone with little knowledge or context.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
Note: the age of the person to whom you're explaining your learning concept can vary. Focus on being able to explain something you understand to someone with little knowledge or context.
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 25]
```diff
@@ -0,0 +1,259 @@
...
+
+The second technique I want to point out is called [Ultralearning](https://www.scotthyoung.com/blog/ultralearning/). In his book, Scott Young explains Ultralearning as,
+
+> a strategy for aggressive, self-directed learning.
+>
+> —Scott Young
+
+Think of it as taking something, learning quickly, and diving deep. You can use this with your learning projects.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
Think of it as taking something, being aggressive with how much or how fast you can learn it, and diving deep. You can use this with your learning projects.
```

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 26]
```diff
@@ -0,0 +1,259 @@
...
+
+The idea is that if you can't do that, you don't know the concept well enough.
+
+Note: the age of the person towhom you're explaining your learning concept can vary. The goal is for you to be able to explain something you understand to someone with little knowledge or context.
+
+The second technique I want to point out is called [Ultralearning](https://www.scotthyoung.com/blog/ultralearning/). In his book, Scott Young explains Ultralearning as,
+
+> a strategy for aggressive, self-directed learning.
+>
+> —Scott Young
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]: Should I remove this?

Wondering if Fast Framework is really my own version of ultralearning.

File: content/2020-06-19-how-to-learn-typescript-quickly/index.md [hunk 27]
```diff
@@ -0,0 +1,259 @@
...
+
+Think of it as taking something, learning quickly, and diving deep. You can use this with your learning projects.
+
+While you're learning, take notes of questions you have or concepts you don't quite understand. Then at the end of your project, see if those questions remain. If they do, start a new learning project and focus on answering those questions.
+
+As you can imagine, if you do this with a topic, you'll have several learning projects and soon start to develop **expertise** in this subject. It's very powerful.
+
+## Summary
+
+The most important takeaway is **use systems to help you learn better and faster**. They will serve as your map, keep you on track and help you find the flag to finish the race. Following these steps with TypeScript in mind, you will develop a proficiency, hopefully even **expertise** in the language.
```
Comment (jsjoeio) [inline content/2020-06-19-how-to-learn-typescript-quickly/index.md]:
```suggestion
The most important takeaway is this: **use systems to help you learn better and faster**. They will serve as your map, keep you on track and help you find the flag to finish the race. Following these steps with TypeScript in mind, you will develop a proficiency, hopefully even **expertise** in the language.
```

**Comment by jsjoeio** [comment]
From @seanjun21

> ...By the way, I won’t critique on the content. That’s your value and personal opinion, I don’t want to touch that. I will just critique on the overall architecture.
>
> 1. I think you could “upgrade” your title to encompass bigger concept — like instead of TypeScript, your title could be about > “learn a programming language fast.” 
>
> Because in the text, you literally said you are going to use TypeScript as an example.
>
> 2. Maybe you could include underline for big 3 titles (Make your Map / Stay on Track / Find the Flag) for viewers reading this > on the phone. The texts are long and titles are kind of hard to follow (since there are so many of them). 
> 
> If you highlight the big three (underline or different color, different font??), I think the viewer can have a general sense of where they are reading. 
>
> 3. Maybe a little of pictures could help here and there? Some photos that are related to content. Like “share with others (yoda quote)” could be supplemented with actual Yoda meme.

**Comment by jsjoeio** [comment]
From @sw-yx 
> I'd say this is a good general "how to learn anything" post with some typescript examples. would be nice to have more typescript specific segments. like if I removed all the typescripty bulletpoints, does it still read like a typescript article? should it? basically there's this tradeoff between general applicability and typescript specificity. I think either is fine but not both at once. know what I mean?

**Comment by jsjoeio** [comment]
Feedback @tadscritch 

> Are Timeline and Timeframe too similar?

Also, check those excalidraw drawings.

**Comment by jsjoeio** [comment]
image

Fix these bullet points lists

**Comment by maxcell** [comment]
> on Twitter, using the #100DaysOfCode hashtag or #CodeNewbie

Maybe: on Twitter, using the #100DaysOfCode or #CodeNewbie hashtags

**Comment by jsjoeio** [comment]
> Feedback @tadscritch
> 
> > Are Timeline and Timeframe too similar?
> 
> Also, check those excalidraw drawings.

Clarified a bit
